src/mgmt/srv.c

```diff
@@ -429,6 +431,146 @@ void ds_mgmt_pool_find_hdlr(crt_rpc_t *rpc)
 	d_rank_list_free(out->pfo_ranks);
 }
 
+static int
+check_cred_pool_access(uuid_t pool_uuid, d_rank_list_t *svc_ranks, int flags, d_iov_t *cred)
+{
+	daos_prop_t            *props           = NULL;
+	struct daos_prop_entry *acl_entry       = NULL;
+	struct daos_prop_entry *owner_entry     = NULL;
+	struct daos_prop_entry *owner_grp_entry = NULL;
+	struct d_ownership      owner;
+	uint64_t                sec_capas;
+	int                     rc;
+
+	rc = ds_mgmt_pool_get_acl(pool_uuid, svc_ranks, &props);
+	if (rc != 0) {
+		DL_ERROR(rc, DF_UUID ": failed to retrieve ACL props", DP_UUID(pool_uuid));
+		return rc;
+	}
+
+	acl_entry = daos_prop_entry_get(props, DAOS_PROP_PO_ACL);
+	D_ASSERT(acl_entry != NULL);
+	D_ASSERT(acl_entry->dpe_val_ptr != NULL);
+
+	owner_entry = daos_prop_entry_get(props, DAOS_PROP_PO_OWNER);
+	D_ASSERT(owner_entry != NULL);
+	D_ASSERT(owner_entry->dpe_str != NULL);
+
+	owner_grp_entry = daos_prop_entry_get(props, DAOS_PROP_PO_OWNER_GROUP);
+	D_ASSERT(owner_grp_entry != NULL);
+	D_ASSERT(owner_grp_entry->dpe_str != NULL);
+
+	owner.user  = owner_entry->dpe_str;
+	owner.group = owner_grp_entry->dpe_str;
+
+	rc = ds_sec_pool_get_capabilities(flags, cred, &owner, acl_entry->dpe_val_ptr, &sec_capas);
+	if (rc != 0) {
+		DL_ERROR(rc, DF_UUID ": failed to read sec capabilities", DP_UUID(pool_uuid));
+		D_GOTO(out_props, rc);
+	}
+
+	if (!ds_sec_pool_can_connect(sec_capas)) {
+		D_GOTO(out_props, rc = -DER_NO_PERM);
+	}
+
+out_props:
+	daos_prop_free(props);
+	return rc;
+}
+
+void
+ds_mgmt_pool_list_hdlr(crt_rpc_t *rpc)
+{
+	struct mgmt_pool_list_in   *in;
+	struct mgmt_pool_list_out  *out;
+	size_t                      n_mgmt = 0, n_rpc = 0;
+	daos_mgmt_pool_info_t      *mgmt_pools = NULL;
+	struct mgmt_pool_list_pool *rpc_pools  = NULL;
+	int                         i, rc, chk_rc;
+
+	in = crt_req_get(rpc);
+	D_ASSERT(in != NULL);
+
+	out = crt_reply_get(rpc);
+	D_ASSERT(out != NULL);
+
+	if (in->pli_npools > 0) {
+		D_ALLOC_ARRAY(mgmt_pools, in->pli_npools);
+		if (mgmt_pools == NULL) {
+			D_ERROR("failed to alloc mgmt pools\n");
+			D_GOTO(send_resp, rc = -DER_NOMEM);
+		}
+	}
+
+	n_mgmt = in->pli_npools;
+	rc     = ds_get_pool_list(&n_mgmt, mgmt_pools);
+	if (rc != 0) {
+		DL_ERROR(rc, "ds_get_pool_list() failed");
+		D_GOTO(send_resp, rc);
+	}
+
+	/* caller just needs the number of pools */
+	if (in->pli_npools == 0) {
+		out->plo_npools = n_mgmt;
+		D_GOTO(send_resp, rc);
+	}
+
+	D_ALLOC_ARRAY(rpc_pools, n_mgmt);
+	if (rpc_pools == NULL) {
+		D_ERROR("failed to alloc response pools\n");
+		D_GOTO(send_resp, rc = -DER_NOMEM);
+	}
+
+	for (i = 0; i < n_mgmt; i++) {
+		daos_mgmt_pool_info_t      *mgmt_pool = &mgmt_pools[i];
+		struct mgmt_pool_list_pool *rpc_pool  = &rpc_pools[n_rpc];
+		chk_rc = check_cred_pool_access(mgmt_pool->mgpi_uuid, mgmt_pool->mgpi_svc,
```

Comment:
nit - In coding guidelines we are supposed to include a blank line after variable declaration section.

Reply:
Done.